client/reducers/action.js

```diff
@@ -11,23 +11,28 @@ export const QUEUE_ACTION = 'QUEUE_ACTION'
 export const HAS_FINISHED = 'HAS_FINISHED'
 export const LEVEL_WON = 'LEVEL_WON'
 export const REMOVE_ACTION = 'REMOVE_ACTION'
+// export const DECREASE_MOVES_LEFT = 'DECREASE_MOVES_LEFT'
 
 export const runCommands = () => {
   return (dispatch, getState) => {
     var interval = setInterval(() => {
       var state = getState()
-      if (state.running === false || state.robot.isAlive == false) {
+      // If robot is not running or dead or moveLimit runs out, stop movment and change button to retry
+      if (!state.running || !state.robot.isAlive || state.moveLimit === state.executeCommandIndex) {
+        dispatch({type: HAS_FINISHED})
```

Comment:
The dispatch of has_finished is what the test is crying about.

The test sets running to false and it expects no more dispatches from runCommands after that.